src/main/java/com/minecolonies/coremod/blocks/BlockMinecoloniesRack.java

```diff
@@ -129,79 +138,97 @@ public IBlockState getActualState(final IBlockState state, final IBlockAccess wo
         else
         {
-            if(rack.getOtherChest() != null)
+            if (rack.getOtherChest() != null)
             {
-                if(rack.isMain())
+                if (rack.isMain())
                 {
-                    return state.withProperty(BlockMinecoloniesRack.VARIANT, EnumType.FULLDOUBLE)
-                            .withProperty(FACING, BlockPosUtil.getFacing(rack.getNeighbor(), pos));
+                    return state.withProperty(BlockMinecoloniesRack.VARIANT, RackType.FULLDOUBLE)
+                             .withProperty(FACING, BlockPosUtil.getFacing(rack.getNeighbor(), pos));
                 }
                 else
                 {
-                    return state.withProperty(BlockMinecoloniesRack.VARIANT, BlockMinecoloniesRack.EnumType.EMPTYAIR);
+                    return state.withProperty(BlockMinecoloniesRack.VARIANT, RackType.EMPTYAIR);
                 }
             }
             else
             {
-                return state.withProperty(BlockMinecoloniesRack.VARIANT, EnumType.FULL);
+                return state.withProperty(BlockMinecoloniesRack.VARIANT, RackType.FULL);
             }
         }
     }
 
     /**
-     * Check if a certain block should be replaced with a rack.
-     * @param block the block to check.
-     * @return true if so.
+     * Convert the BlockState into the correct metadata value.
+     *
+     * @deprecated (Remove this as soon as minecraft offers anything better).
      */
-    public static boolean shouldBlockBeReplacedWithRack(final Block block)
+    @NotNull
+    @Override
+    @Deprecated
+    public IBlockState withRotation(@NotNull final IBlockState state, final Rotation rot)
     {
-        return block == Blocks.CHEST || block == ModBlocks.blockRack;
+        return state.withProperty(FACING, rot.rotate(state.getValue(FACING)));
     }
 
+    /**
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @NotNull
     @Override
-    public int getMetaFromState(@NotNull final IBlockState state)
+    @Deprecated
+    public IBlockState withMirror(@NotNull final IBlockState state, final Mirror mirrorIn)
     {
-        return state.getValue(FACING).getHorizontalIndex();
+        return state.withRotation(mirrorIn.toRotation(state.getValue(FACING)));
+    }
+
+    /**
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @Override
+    @Deprecated
+    public boolean isFullCube(final IBlockState state)
+    {
+        return false;
     }
 
     /**
-     * Does some action if a neighbor changes.
-     * @deprecated but we definitely need it.
-     * @param state the state.
-     * @param worldIn the world.
-     * @param pos the pos.
-     * @param blockIn the block.
-     * @param fromPos the changed ps.
+     * @deprecated (Remove this as soon as minecraft offers anything better).
      */
     @Override
     @Deprecated
+    public boolean isOpaqueCube(final IBlockState state)
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)